src/quantile_rank_sorted.js

```diff
@@ -0,0 +1,93 @@
+/* @flow */
+/* eslint no-bitwise: 0 */
+
+/**
+ * This function returns the quantile in which one would find the given value in
+ * the given array. With a sorted array, leveraging binary search, we can find
+ * this information in logarithmic time.
+ *
+ * @param {Array<number>} x input
+ * @returns {number} value value
+ * @example
+ * quantileRankSorted([1, 2, 3, 4], 3); // => 0.75
+ * quantileRankSorted([1, 2, 3, 3, 4], 3); // => 0.7
+ * quantileRankSorted([1, 2, 3, 4], 6); // => 1
+ * quantileRankSorted([1, 2, 3, 3, 5], 4); // => 0.8
+ */
+function quantileRankSorted(
+    x /*: Array<number> */,
+    value /*: number */)/*: number */ {
+
+    var n = x.length;
+
+    var l = lowerBound(x, value);
+
+    // Value is lesser than any value in the array
```

Comment:
From my reading, we can do these early-returns _before_ running `lowerBound`, which for those cases would save a few cycles.

Reply:
Correct. I'll update this soon.